vscode-extension/src/util.ts

```diff
@@ -81,6 +81,12 @@ export function updateWebviewEditorThemeMode(webview: vscode.Webview) {
   });
 }
 
+export function getConfigurationTarget() {
+  return vscode.workspace.workspaceFolders !== undefined
+    ? vscode.ConfigurationTarget.Workspace
+    : vscode.ConfigurationTarget.Global;
```

Comment:
Actually yea, don't we also need to add it https://github.com/lastmile-ai/aiconfig/blob/2b9d1c651c366a15c82bebcb32dee8406dfd624e/vscode-extension/src/extension.ts#L440 otherwise this will fail in the use case where we want to define a model parser registry when the workspace path hasn't been set yet?

Reply:
Also not related to this issue, but does this command to register do anything? https://github.com/lastmile-ai/aiconfig/blob/2b9d1c651c366a15c82bebcb32dee8406dfd624e/vscode-extension/src/extension.ts#L414 The variable `modelRegistryPath` is never used or saved later in this function

Reply:
> The variable modelRegistryPath is never used or saved later in this function

It does get saved in here in that function:
```
await handleCustomModelRegistryUpdate(
    config,
    aiconfigEditorManager,
    modelRegistryPath
  );
```

> otherwise this will fail in the use case where we want to define a model parser registry when the workspace path hasn't been set yet?

I think this is ok -- basically `getConfiguration` uses the lowest applicable scope to get the configuration from. If it can't find it in user, then it checks workspace, then checks global

> Nice, does this also ensure that we don't write anything to the .vscode/settings.json file?

Not sure about this -- we can look into this as a follow-up :)